mypy/errorcodes.py

```diff
@@ -273,6 +273,14 @@ def __hash__(self) -> int:
 # This is a catch-all for remaining uncategorized errors.
 MISC: Final[ErrorCode] = ErrorCode("misc", "Miscellaneous other checks", "General")
 
+OVERLOAD_CANNOT_MATCH: Final[ErrorCode] = ErrorCode(
+    "overload-cannot-match",
+    "Warn user about signature matching for overloaded functions.",
```

Comment:
```suggestion
    "Warn if an @overload signature can never be matched",
```
Consistent formatting with the next one.